app/models/container_template.rb

```diff
@@ -34,11 +34,23 @@ def instantiate(params, project = nil)
                                             :namespace => project
                                           },
                                           :objects    => objects,
-                                          :parameters => params)
+                                          :parameters => params_to_hashes(params))
     create_objects(processed_template['objects'], project)
     @created_objects.each { |obj| obj[:miq_class] = MIQ_ENTITY_MAPPING[obj[:kind]] }
   end
 
+  def params_to_hashes(params)
+    params.collect do |param|
+      {
+        :name     => param.name,
+        :value    => param.value,
+        :generate => param.generate,
+        :from     => param.from,
+        :required => param.required
+      }
+    end
+  end
+
```

Comment:
That's what `:parameters` is expecting, yeah. Doesn't seem like the worst name; the more annoying part (that's beyond the scope of this change) is overloading the term `params` (used everywhere in a Rails app as a hash of parameters) with an array of ContainerTemplateParameters :)